### ISSUE TITLE
Deprecate GWT.create .... move to annotation processing.

### ISSUE THREAD

**Issue by muksihs** [google/gwteventbinder#33]
Are there any plans, beta code yet, to move to using only annotation processing in combination with interfaces to do something similar to current EventBinder capabilities?

**Comment by ekuefler** [comment]
Duplicate of #13

**Comment by ekuefler** [comment]
As far as I know, GWT.create is still supported and there aren't official plans yet in how it will be replaced in GWT 3.0. Once we learn more about how GWT 3.0 will work it shouldn't be hard to update gwteventbinder accordingly.

**Comment by niloc132** [comment]
I'm aware of no plans (either from Google or any other team working to build GWT 3) to allow GWT.create calls to continue working in GWT 3. Instead, the word has been that all Generators should be rewritten into Annotation Processors (and all JSNI/JSOs remade into JsInterop, etc).

At present, GWT 3 is "mostly" J2CL plus Closure Compiler. While much of the TypeOracle and JType hierarchy can be updated to use APT tools, there are some whole world aspects that will not work correctly. Since J2CL isn't itself "whole-world" (closure takes on that role), it isn't possible to reliably generate Java with knowledge of all other Java that will eventually exist in the app. Depending on the application, it might be possible to impose limitations from certain generators (like mandating that all processor runs happen just before j2cl runs, and then that closure will run only on those transpiled sources (and no other later transpiled java).

**Comment by lofidewanto** [comment]
If I'm not wrong: is there only these two generator files which needed to rewrite using Java Annotation Processor?

https://github.com/google/gwteventbinder/tree/master/eventbinder/src/main/java/com/google/web/bindery/event/gwt/rebind/binder

Thanks,
Lofi

**Comment by michael-conrad** [comment]
really looks that way.

**Comment by ekuefler** [comment]
Yep, creating a version of EventBinder using annotation processors should be pretty trivial. I definitely support anyone who wants to start a separate project now to pursue this, but am planning to wait for published formal guidance around GWT 3.0 before making any major changes here.
